### DISCUSSION TITLE
Scroll only when text exceeds max width

### DISCUSSION THREAD

**Discussion by Kellsen** [brndnmtthws/conky#1803] [Ideas]
Heya,

I have my Conky set up so it displays what's playing on my music player (deadbeef).
So I have a max width set to prevent the Conky to go "out of bounds". When i.e. the song title exceeds the max width it now obviously gets cut off.
When I set the ${scroll} variable it does what I want it to do but it does that with all titles even if they would fit the max width which I find a bit irritating.

Is there an option to set the ${scroll} only for the titles that exceed the max width and don't scroll for text that fits the boundaries?

See the screenshots for the problem.

Cheers!

![Bildschirmfoto vom 2024-04-06 17-39-47](https://github.com/brndnmtthws/conky/assets/93977235/135f975b-99dc-4bc1-9990-101530a845a5)
![Bildschirmfoto vom 2024-04-06 17-41-02](https://github.com/brndnmtthws/conky/assets/93977235/72b9796c-43c7-4c34-8e05-2bf700f0c4da)

**Comment by Caellian** [comment]
I don't think there is (though it would be useful).

You can achieve what you're looking for from lua though by:
- Figuring out bounding coordinates (given that the position is fixed (i.e. no variable height content above))
- Testing text width with [`cairo-text-extents`](https://www.cairographics.org/manual/cairo-text.html#cairo-text-extents)
  - If it can fit, render it normally
  - Else, render with special _render function_

Special _render function_ takes current `time`, `text` and `bounds`:
- render text onto a separate surface with x position adjusted by some interpolation function
- apply effects (e.g. fade on edges)
- copy a fixed part of that surface onto main ([example](https://cairo.cairographics.narkive.com/AHWZRS9D/how-to-copy-a-part-from-one-surface-to-another))

I'm sorry my explanation is sparse, I'm a bit short on time. Feel free to mention me if you _do_ decide to implement it yourself and get stuck, I'll reply asap.

**Reply by Kellsen** [reply]
Alright, cheers for your answer :+1: 
Tbh I can't be bothered. I wrote quite a few scripts for this Conky already and I have no idea what I'm doing in the first place. Just stitched together with zero proper knowledge. I'm surprised it's working at all :smile: 
But if you agree it would be a nice feature maybe we can put that in the "Requests" section?

**Reply by Caellian** [reply]
Moved the discussion to Ideas. TBH, there's a whole bunch of work left for Wayland support so this I probably not getting implemented that soon unless someone creates a PR.

If you end up implementing it in Lua, do share because Conky also uses cairo internally so it'd probably help out the person implementing this.

**Comment by bebeidon** [comment]
can you share your config please? i would love a media monitor like that :)
